keras/src/quantizers/quantizers_test.py

```diff
@@ -100,3 +100,759 @@ def test_quantize_and_dequantize(self):
         )
         # A loose assertion due to an expected quantization error
         self.assertAllClose(qdq_values, values, atol=5e-1)
+
+    def _TestOp(
+        self,
+        op,
+        input_min,
+        input_max,
+        num_bits,
+        narrow_range,
+        expected_nudged_input_min,
+        expected_nudged_input_max,
+        expected_step,
+    ):
+        inputs = ops.array(
+            [
+                expected_nudged_input_min - expected_step,
+                expected_nudged_input_min - 0.01,
+                expected_nudged_input_min,
+                expected_nudged_input_min + 0.01,
+                expected_nudged_input_min + expected_step - 0.01,
+                expected_nudged_input_min + expected_step,
+                expected_nudged_input_min + expected_step + 0.01,
+                expected_nudged_input_max - 0.01,
+                expected_nudged_input_max,
+                expected_nudged_input_max + 0.01,
+                expected_nudged_input_max + expected_step,
+            ],
+            dtype="float32",
+        )
+        expected = ops.array(
+            [
+                expected_nudged_input_min,
+                expected_nudged_input_min,
+                expected_nudged_input_min,
+                expected_nudged_input_min,
+                expected_nudged_input_min + expected_step,
+                expected_nudged_input_min + expected_step,
+                expected_nudged_input_min + expected_step,
+                expected_nudged_input_max,
+                expected_nudged_input_max,
+                expected_nudged_input_max,
+                expected_nudged_input_max,
+            ],
+            dtype="float32",
+        )
+
+        outputs = op(
+            inputs,
+            input_min,
+            input_max,
+            num_bits=num_bits,
+            narrow_range=narrow_range,
+        )
+        self.assertAllClose(outputs, expected)
```

Comment:
I think verifying the output values alone is not sufficient.
We can add an assertion for the gradient, similar to this one:
https://github.com/keras-team/keras/blob/master/keras/src/layers/core/dense_test.py#L584